include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Dim username As String = "admin"
Dim password As String = "Password123" ' Noncompliant
Dim usernamePassword As String = "user=admin&password=Password123" ' Noncompliant
Dim url As String = "scheme://user:Admin123@domain.com" ' Noncompliant
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim username As String = "admin"
Dim password As String = GetEncryptedPassword()
Dim usernamePassword As String = String.Format("user={0}&password={1}", GetEncryptedUsername(), GetEncryptedPassword())
Dim url As String = $"scheme://{username}:{password}@domain.com"

Dim url2 As String= "http://guest:guest@domain.com" ' Compliant
Const Password_Property As String = "custom.password" ' Compliant
----

== Exceptions

* Issue is not raised when URI username and password are the same.
* Issue is not raised when searched pattern is found in variable name and value.

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
