== Why is this an issue?

include::../description.adoc[]

=== Exceptions

To prevent generating some false-positives, literals having less than 10 characters are excluded as well as literals matching ``++/^\w*$/++``.

The following patterns are also excluded based on AST context rather than string content:

* Import and export paths (e.g., `import x from 'path'`)
* `require()` calls
* JSX attributes
* Type declarations and object properties
* Standalone string statements like directives (e.g., `'use strict';`)

The `ignoreStrings` parameter is additive to these built-in exceptions, allowing you to specify additional strings to ignore via comma-separated values.

== How to fix it

include::../howtofix.adoc[]

=== Code examples

==== Noncompliant code example

With the default threshold of 3:

[source,javascript,diff-id=1,diff-type=noncompliant]
----
function run() {
    prepare("action_to_launch");  // Noncompliant - "action_to_launch" is duplicated 3 times
    execute("action_to_launch");
    release("action_to_launch");
}

function printInQuotes(a, b) {
  console.log("'" + a + "'" + b + "'");               // Compliant - literal "'" has less than 10 characters and is excluded
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
var ACTION_1 = "action_to_launch";

function run() {
  prepare(ACTION_1);                               // Compliant
  execute(ACTION_1);
  release(ACTION_1);
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''

endif::env-github,rspecator-view[]
