include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: app
spec:
  containers:
    - name: my-container
      env:
        - name: password
          value: f68rth6sd5g61gs5  # Noncompliant
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: app
spec:
  containers:
    - name: my-container
      env:
        - name: password
          valueFrom:
            secretKeyRef:
              name: my-secret
              key: password-key
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
