:detections: variables/fields
:defaultsensibility: 3

include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
let secret = "47828a8dd77ee1eb9dde2d5e93cb221ce8c32b37" // Noncompliant

func main() {
    callMyService(secret)
}
----

==== Compliant solution

If you are building a mobile application, it is recommended to keep API keys in a secure backend instead of hardcoding them in the mobile application.
Requests to an API can then be proxied from the mobile application through that backend.

In order to communicate with the API, you can set up an authentication flow with the backend. The resulting access key should be kept in the Keychain:

[source,swift,diff-id=1,diff-type=compliant]
----
func storeSecret(accessKey: String) throws {
    let accessControl = SecAccessControlCreateWithFlags(
     nil,
     kSecAttrAccessibleWhenUnlockedThisDeviceOnly,
     .biometryCurrentSet,
     nil)

    let query: [CFString: Any] = [
        kSecClass: kSecClassGenericPassword,
        kSecAttrService: "com.example.app",
        kSecAttrAccount: "Example account",
        kSecAttrAccessControl: accessControl as Any,
        kSecValueData: accessKey.data(using: .utf8)!
    ]
                
    SecItemAdd(query as CFDictionary, nil)
}
----


If you are building a backend application instead, it is still recommended to use a secure solution to store API keys.

One such example is https://sdk.amazonaws.com/swift/api/awssecretsmanager/latest/documentation/awssecretsmanager[AWS Secrets Manager with the AWS SDK for Swift]:

[source,swift]
----
import AWSSecretsManager

func main() async throws {
    let secretsClient = try SecretsManagerClient()

    let input = GetSecretValueInput(secretId: "MY_SERVICE_SECRET")
    let response = try await secretsClient.getSecretValue(input)
    if let secret = response.secretString {
        callMyService(secret)
    }
}
----

include::../resources.adoc[]

* OWASP - https://owasp.org/www-project-mobile-top-10/2023-risks/m1-improper-credential-usage.html[Mobile Top 10 2024 Category M1 - Improper Credential Usage]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]


include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
