:detections: variables/fields
:defaultSensibility: 3

include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
var secret = "47828a8dd77ee1eb9dde2d5e93cb221ce8c32b37" // Noncompliant

func main() {
    callMyService(secret)
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import "os"

var secret = os.Getenv("SECRET")

func main() {
    callMyService(secret)
}
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]


include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
