include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
username: "root", # Noncompliant
password: "qwerty123", # Noncompliant
url: "https://example.com/login?user=root&password=qwerty123" # Noncompliant
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
username: "root",
password: "${{ secrets.MY_APP_PASSWORD }}",
url: "https://example.com/login?user=root&password=${{ secrets.MY_APP_PASSWORD }}"
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
