== Why is this an issue?

Unused variables are variables that are assigned but never read. Such assignments are unnecessary and don't contribute to the functionality or clarity of the code. They may even negatively impact performance. Removing them enhances code cleanliness and readability. Even if the unnecessary operations do not do any harm in terms of the program's correctness, they are - at best - a waste of computing resources.

=== Exceptions

This rule ignores initializations to `-1`, `0`, `1`, `nil`, `true`, `false` and `""`.

== How to fix it

Remove the unnecessary assignment, then test the code to make sure that the right-hand side of a given assignment had no side effects (e.g. a method that writes certain data to a file and returns the number of written bytes).

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
def calculate(a, b)
  i = a + b  # Noncompliant: variable is never read
  42
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
def calculate(a, b)
  i = a + b
  i
end
----

include::../see.adoc[]

=== Related rules

* S1763 - All code should be reachable
* S3516 - Functions returns should not be invariant
* S3626 - Jump statements should not be redundant

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
