include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const mysql = require('mysql');

const connection = mysql.createConnection({
  host:'localhost',
  user: "admin",
  database: "project",
  password: "mypassword", // Noncompliant
  multipleStatements: true
});

connection.connect(); 
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const mysql = require('mysql');

const connection = mysql.createConnection({
  host: process.env.MYSQL_URL,
  user: process.env.MYSQL_USERNAME,
  password: process.env.MYSQL_PASSWORD,
  database: process.env.MYSQL_DATABASE
});
connection.connect();
----

include::../resources.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Review this potentially hard-coded password.


=== Parameters

.passwordWords
****

----
password, passwd, pwd, passphrase
----

Comma separated list of words identifying potential password
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
