:detections: variables/fields/properties
:defaultSensibility: 3

include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Private MySecret As String = "47828a8dd77ee1eb9dde2d5e93cb221ce8c32b37"
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Private MySecret As String  = Environment.GetEnvironmentVariable("MYSECRET")
----

include::../resources.adoc[]

* MSC - https://wiki.sei.cmu.edu/confluence/x/OjdGBQ[MSC03-J - Never hard code sensitive information] 

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
