include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "username": "root",
  "password": "qwerty123",
  "url": "https://example.com/login?user=root&password=qwerty123"
}
----

==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "username": "root",
  "password": "${MY_APP_PASSWORD}",
  "url": "https://example.com/login?user=root&password=${MY_APP_PASSWORD}"
}
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
