include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,tsql,diff-id=1,diff-type=noncompliant]
----
DECLARE @password nvarchar(100) = 'hardcoded_pwd'; -- Noncompliant
----

==== Compliant solution

[source,tsql,diff-id=1,diff-type=compliant]
----
DECLARE @password nvarchar(100) = dbo.GetDatabasePassword(); -- Compliant
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
