:detections: variables/fields
:defaultSensibility: 5

include::../description.adoc[]

=== Code Examples

==== Noncompliant Code Example

[source,php,diff-id=1,diff-type=noncompliant]
----
$secret = '47828a8dd77ee1eb9dde2d5e93cb221ce8c32b37'; // Noncompliant
MyClass->callMyService($secret);
----

==== Compliant solution

Store the secret in an environment variable or configuration file:

[source,php,diff-id=1,diff-type=compliant]
----
$secret = getenv('MY_SECRET');
MyClass->callMyService($secret);
----

Or using https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/php/example_code/secretsmanager[AWS Secrets Manager]:

[source,php]
----
use Aws\SecretsManager\SecretsManagerClient;
use Aws\Exception\AwsException;
$client = new SecretsManagerClient(...);
$secretName = 'example';
doSomething($client, $secretName)
function doSomething($client, $secretName) {
    try {
        $result = $client->getSecretValue([
            'SecretId' => $secretName,
        ]);
    } catch (AwsException $e) {
    ...
    }
    if (isset($result['SecretString'])) {
        $secret = $result['SecretString'];
    } else {
        $secret = base64_decode($result['SecretBinary']);
    }
    // do something with the secret
    MyClass->callMyService($secret);
}
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
