include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
username = 'admin'
password = 'admin' # Noncompliant
usernamePassword = 'user=admin&password=admin' # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import os

username = os.getenv("username") # Compliant
password = os.getenv("password") # Compliant
usernamePassword = 'user=%s&password=%s' % (username, password) # Compliant
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
