:detections: variables/fields
:defaultsensibility: 5

include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
private val MY_SECRET = "47828a8dd77ee1eb9dde2d5e93cb221ce8c32b37"

fun main() {
  MyClass.callMyService(MY_SECRET)
}
----

==== Compliant solution

Using https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/javav2/example_code/secrets-manager[AWS Secrets Manager]:

[source,kotlin,diff-id=1,diff-type=compliant]
----
import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;

fun main() {
  SecretsManagerClient secretsClient = ...
  MyClass.doSomething(secretsClient, "MY_SERVICE_SECRET")
}

fun doSomething(secretsClient: SecretsManagerClient, secretName: String) {
  val valueRequest = GetSecretValueRequest.builder()
    .secretId(secretName)
    .build()

  val valueResponse = secretsClient.getSecretValue(valueRequest)
  val secret = valueResponse.secretString()
  // do something with the secret
  MyClass.callMyService(secret)
}
----

Using https://docs.microsoft.com/en-us/azure/key-vault/secrets/quick-create-java?tabs=azure-cli[Azure Key Vault Secret]:

[source,kotlin]
----
import com.azure.identity.DefaultAzureCredentialBuilder;

import com.azure.security.keyvault.secrets.SecretClient;
import com.azure.security.keyvault.secrets.SecretClientBuilder;
import com.azure.security.keyvault.secrets.models.KeyVaultSecret;

fun main() {
  val keyVaultName = System.getenv("KEY_VAULT_NAME")
  val keyVaultUri = "https://$keyVaultName.vault.azure.net"

  val secretClient = SecretClientBuilder()
    .vaultUrl(keyVaultUri)
    .credential(DefaultAzureCredentialBuilder().build())
    .buildClient()

  MyClass.doSomething(secretClient, "MY_SERVICE_SECRET")
}

fun doSomething(secretClient: SecretClent, secretName: String) {
  val retrievedSecret = secretClient.getSecret(secretName)
  val secret = retrievedSecret.getValue()

  // do something with the secret
  MyClass.callMyService(secret)
}
----


include::../resources.adoc[]

* OWASP - https://owasp.org/www-project-mobile-top-10/2023-risks/m1-improper-credential-usage.html[Mobile Top 10 2024 Category M1 - Improper Credential Usage]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]


include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
