include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,scala,diff-id=1,diff-type=noncompliant]
----
val password = "xxxx" // Noncompliant
----

==== Compliant solution

[source,scala,diff-id=1,diff-type=compliant]
----
val password = retrievePassword()
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
