:detections: keys
:defaultSensibility: 3

include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: nginx-app
spec:
  containers:
    - name: nginx
      image: "nginx:1.21.6"
      ports:
        - containerPort: 80
      env:
        - name: API_TOKEN
          value: "f7a9s8d7f6as98df7a6s9d8f7a6sd9f87as6df"  # Noncompliant
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: nginx-app
spec:
  containers:
    - name: nginx
      image: "nginx:1.21.6"
      ports:
        - containerPort: 80
      env:
        - name: API_TOKEN
          valueFrom:
            secretKeyRef:
              name: my-secret
              key: api-key
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]


include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
