:detections: variables/fields
:defaultSensibility: 5

include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
char const *mySecret = "47828a8dd77ee1eb9dde2d5e93cb221ce8c32b37";
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
const char* mySecret = std::getenv("MY_APP_SECRET");
----

include::../resources.adoc[]


ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
