include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,abap,diff-id=1,diff-type=noncompliant]
----
DATA: password(10) VALUE 'secret123',
           pwd(10) VALUE 'secret123'.
----

==== Compliant solution

[source,abap,diff-id=1,diff-type=compliant]
----
DATA: password(10) TYPE string,
           pwd(10) TYPE string.
password = get_encrypted_password( ).
pwd = get_encrypted_password( ).
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
