include::../why.adoc[]

=== Exceptions

* The rule ignores string literals that are used directly in Regexp methods.

include::../how-to-fix-it.adoc[]

=== Code Examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func connect()  {
  user := "root"
  password:= "supersecret" // Noncompliant

  url := "login=" + user + "&passwd=" + password
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func connect()  {
  user := getEncryptedUser()
  password:= getEncryptedPass() // Compliant

  url := "login=" + user + "&passwd=" + password
}
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
