:detections: variables/fields
:defaultSensibility: 5

include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const API_KEY = "1234567890abcdef"  // Noncompliant

const response = await fetch("https://api.my-service/v1/users", {
  headers: {
    Authorization: `Bearer ${API_KEY}`,
  },
});
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const API_KEY = process.env.API_KEY;

const response = await fetch("https://api.my-service/v1/users", {
  headers: {
    Authorization: `Bearer ${API_KEY}`,
  },
});
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]


include::../parameters.adoc[]

'''
endif::env-github,rspecator-view[]
