include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
password = 'mypassword' # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
password = retrieve_password()
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
