include::../description.adoc[]

=== Code Examples

==== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
let postData = "username=Steve&password=123456".data(using: .utf8)  // Noncompliant
//...
var request = URLRequest(url: url)
request.HTTPBody = postData
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
let postData = "username=\(getEncryptedUser())&password=\(getEncryptedPass())".data(using: .utf8)
//...
var request = URLRequest(url: url)
request.HTTPBody = postData
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
